docs/docs/components/migrating/README.md

```diff
@@ -0,0 +1,190 @@
+
+**Develop locally and host on Github**
+Actions are no longer developed in Pipedream's UI. Develop actions locally using your preferred editor, publish to Pipedream via CLI and maintain the code in your own Github repo.
+
+**Update with a click**
+When you publish a new version of an action, you can update actions used in workflows with a click (updating legacy actions in workflows requires action steps to be deleted, re-added and re-configured).
+
+**Support for async options**
+Async options allow action authos to render a paginated drop down menu allowing users to select from vaues that are programatically generated. The most common use case is to populate the drop down based on results of an API request (e.g., to list Google Sheets in a user's drive).
+
+**Simplified discovery**
+Actions you publish are now grouped under **My Actions** when adding a step to a workflow. NOTE: this option will appear in the workflow builder *after* you publish your first action.
+
+## Getting Started
+
+Ready to develop your first component action? We recommend starting with our [quickstart guide](/components/quickstart/nodejs/actions/). Then review both our [component API reference](/components/api) and [actions published to Pipedream’s Github repo](https://github.com/pipedreamhq/pipedream/compnents).
+
+## Migration Example
+
+Let's walk through an example that migrates code for a legacy action to a Pipedream component. 
+
+### Legacy Code Example
+
+Following is the code for the legacy action to get a Github repo (Github was linked to this action via Pipedream's UI, so it's not declared in the code):
+
+```javascript
+const config = {
+  url: `https://api.github.com/repos/${params.owner}/${params.repo}`,
+  headers: {
+    Authorization: `Bearer ${auths.github.oauth_access_token}`,
+  }
+}
+return await require("@pipedreamhq/platform").axios(this, config)
+```
+
+Also, following is the associated JSON schema that defines metadata for the `params` inputs:
+
+```json
+{
+  "type": "object",
+  "properties": {
+    "owner": {
+      "type": "string",
+      "description": "Name of repository owner."
+    },
+    "repo": {
+      "type": "string",
+      "description": "Name of repository."
+    }
+  },
+  "required": [
+    "owner",
+    "repo"
+  ]
+}
+```
+
+### Converting to the Component Model
+
+To convert the code above to the component model, we need to:
+
+1. Link the Github app to the component using `props` (so we can use Pipedream managed auth for Github)
+2. Define `props` for `owner` and `repo` so we can capture user input. The definition for each prop includes the `type` and `description` metadata. Additionally, since all the fields are required, we do not need to set the `optional`  property (set `optional` to `true` for optional `props`). This metadata was previously captured in the JSON schema.
+3. Replace references to `params` in the `run()` method. `props` are bound to `this`. 
```

Comment:
It might help clarify to say "`props` are bound to `this`, e.g., `this.owner` and `this.repo`", just to provide some examples.

Reply:
done